src/resources/views/crud/fields/select_grouped.blade.php

```diff
@@ -32,7 +32,7 @@
                     <optgroup label="{{ $category->{$field['group_by_attribute']} }}">
                         @foreach ($category->{$field['group_by_relationship_back']} as $subEntry)
                             <option value="{{ $subEntry->getKey() }}"
-                                @if ( ( old($field['name']) && old($field['name']) == $subEntry->getKey() ) || (isset($field['value']) && $subEntry->getKey()==$field['value']))
+                                @if ($current_value === $subEntry->getKey())
```

Comment:
Indeed, the problem was not here as I was thinking. I think I found the culprit I am looking at it. 
To sum up:
- the `select_grouped` we have in monsters is a fake field and belongsTo relation with `article` that the foreignKey is `select2_from_ajax` in the monsters table. 
```php
     // the relation
    public function article()
    {
        return $this->belongsTo(\Backpack\NewsCRUD\app\Models\Article::class, 'select2_from_ajax');
    }
        // the field
         [                  
                 'label'                      => 'Select_grouped'.backpack_free_badge(),
                 'type'                       => 'select_grouped', //https://github.com/Laravel-Backpack/CRUD/issues/502
                'name'                       => 'select_grouped_id',
                'fake'                       => true,
                'entity'                     => 'article',
                'model'                      => 'Backpack\NewsCRUD\app\Models\Article',
                'attribute'                  => 'title',
                'group_by'                   => 'category', // the relationship to entity you want to use for grouping
                'group_by_attribute'         => 'name', // the attribute on related model, that you want shown
                'group_by_relationship_back' => 'articles', // relationship from related model back to this model
                'tab'                        => 'Selects',
                'wrapperAttributes'          => ['class' => 'form-group col-md-6'],
          ],
```
is this somehow supposed to work ? when you get the `$monster->article` you always get from the `select2_from_ajax` key . I guess having relations without "actually" having them is kind of unexpected. Fake relations ? What are the benefits ? 

What I think makes sense is if you need the same relation twice is because you need to store the relation twice, example you have two selects for articles, one is `current_article` and the other is `sub_article`, you create two relations `article` and `subArticle` and provide the correct keys for each of them. 

Because ... they are relations rigth ? 

So we either consider this a bad implementation in demo and refactor the field definition, or we go down the hole again to support fake relations. 

Let me know,
Pedro

Reply:
Sorry Pedro, I do not understand the issue. What I understand is that:
- on the 4.2 branch, if you submit the Monster form, it errors and you get back to a form with `select_grouped` and `select2_grouped` pre-filled (old value works);
- on this branch, if you submit the Monster form, it errors and you get back to empty `select_grouped` and `select2_grouped` inputs;

So this change breaks stuff. **What is the purpose of doing this in the first place? Is it just optimization? Or is it fixing an error? How big of an error? How niche of an error?**

It looks to me like we should **remove all changes to `select_grouped` and `select2_grouped` from this branch** - those changes should not have been pushed here. If you want to make changes to these fields, please create a separate PR pointing to this one, explain why the change is needed and we can talk about it there. Talking about it here in this PR is unproductive. This PR has a purpose. This change does not seem to help in that purpose, it is not related.

In the meantime, let's please not let this `select_grouped` proposal for improvement block us from working THE MAIN THING. I feel like this minor issue is holding us back from talking about MUST-DOs. Move this out of the way, please.